runtime/parser/lexer/lexer.go

```diff
@@ -414,6 +424,11 @@ func (l *lexer) scanString(quote rune) {
 				l.backupOne()
 				return
 			}
+		case '$':
+			// string template, stop and set mode
+			l.backupOne()
+			l.mode = STR_IDENTIFIER
+			return
```

Comment:
We might also need a way to escape  the`$` sign when needed. e.g: not all `$` signs in a string are interpolations.
e.g: I may want to keep the $-sign as-is in a string literal such as: `"withdrawn amount: $5.00"`

We could use `\` for escaping, like for other cases. Then I would have to re-write the literal as `"withdrawn amount: \$5.00"`

Reply:
This should already be supported in the lexer and requires a small change in the parser. Added.